corehq/form_processor/submission_post.py

```diff
@@ -140,7 +151,9 @@ def _handle_basic_failure_modes(self):
         if isinstance(self.instance, BadRequest):
             return HttpResponseBadRequest(self.instance.message)
 
-    def _post_process_form(self, xform):
+    def _post_process_form(self, xform, xform_context):
+        for model_obj in xform_context.supplementary_models:
+            xform.track_create(model_obj)
```

Comment:
is there any reason this doesn't happen in directly in the pre-processor instead adding them to `supplementary_models`?

Reply:
`xform` object is not available during pre-processing.